runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -436,6 +427,27 @@
         @Default
         List<MsgHandlingFilter> nettyClientInboundMsgFilters = null;
 
+        /**
+         * The default priority of the requests made by this client.
+         * Under resource constraints non-high priority requests
+         * are dropped.
+         */
+        @Default
+        PriorityLevel priorityLevel = PriorityLevel.NORMAL;
+
+        /**
+         * Port at which the {@link CorfuRuntime} will allow third-party
+         * collectors to pull for metrics.
+         */
+        @Default
+        int prometheusMetricsPort = MetricsUtils.NO_METRICS_PORT;
+
+        /**
+         * The compression codec to use to encode a write's payload
+         */
+        @Default
+        String codecType = Codec.Type.ZSTD.toString();
```

Comment:
Should we disallow changing the codecType. Do we want to support multiple transactions with different codecTypes ?

Reply:
I believe there is no issue with this. I've written a test to confirm that the use of multiple codecs does not imply any issue.